### DISCUSSION TITLE
Add more usage examples!

### DISCUSSION THREAD

**Discussion by ignatenko-paul** [HBOCodeLabs/wiremock-captain#568] [General]
Good day team!
Could you add more examples how to configure mappings with regex on a req/res body?
E.g. 
- how to implement the Request Matching using this library, like:
---------------------
https://wiremock.org/docs/request-matching/
    "bodyPatterns" : [ {
      "equalToJson" : "{ \"total_results\": 4  }",
      "ignoreArrayOrder" : true,
      "ignoreExtraElements" : true
--------------------
- how to implement Response Templating, like:
---------------------
https://wiremock.org/docs/response-templating/
    "response": {
        "body": "{{request.path.[0]}}",
        "transformers": ["response-template"]
    }
---------------------
Thanks in advance.

**Comment by satvik-s** [comment]
hi! thanks for the suggestion. agree that the above would be useful for people trying to use the library.

i am a little busy presently but i will try to get to this as soon as possible

**Reply by ignatenko-paul** [reply]
Thanks for the answer

**Reply by satvik-s** [reply]
@ignatenko-paul if you need these for some urgent use case, i am happy to provide more feedback in the thread if you can share the use case

**Reply by ignatenko-paul** [reply]
I need to match paritally request's body, 
I expect smth like this in my request's body: **body:{'id':'123', 'name':'< regex here >'}**

And also I need to get some value from body and resend it within response, to keep my upstream service in consistency when it mocked..
I expect smth like this in my response's body: **"body": {"name":"< request.name >"}**

**Reply by satvik-s** [reply]
# first use case

the current implementation does not allow doing two checks at the same time. so you can either do

```
mock.register(
  {
      method: 'POST',
      body: "$[?(@.id == '123')]",
      endpoint: '/test-endpoint',
  },
  {
      status: 200,
  },
  {
      requestBodyFeature: MatchingAttributes.MatchesJsonPath,
  },
);
```

or

```
mock.register(
  {
      method: 'POST',
      body: "$[?(@.name =~ /< regex here >/)]",
      endpoint: '/test-endpoint',
  },
  {
      status: 200,
  },
  {
      requestBodyFeature: MatchingAttributes.MatchesJsonPath,
  },
);
```

given the current contract of wiremock-captain, it would require a breaking change to support having both the matching be available at the same time. i will have to think more if we can achieve this without a breaking change but this might be trickier to implement while making minimal change to library API contract

# second use case

it (response templating) is unfortunately not currently supported but this should be easier to extend the library to support. i will try to get to this one soon

**Reply by ignatenko-paul** [reply]
Thanks a lot for such quick answer

**Reply by satvik-s** [reply]
https://github.com/HBOCodeLabs/wiremock-captain/pull/573

will do a release post merging the above which should solve for the second use case

**Reply by satvik-s** [reply]
https://www.npmjs.com/package/wiremock-captain/v/3.5.0
